lib/rubocop/cop/lint/top_level_return_with_argument.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for top level return with arguments. If there is a
+      # top-level return statement with an argument, then the argument is
+      # always ignored. This is detected automatically since Ruby 2.7.
+      #
+      # This cop works by validating the ancestors of the return node. A
+      # top-level return node's ancestors will never belong to `AST::BlockNode`
+      # or `AST::DefNode` class.
+      #
+      # @example
+      #
+      #   # Detected since Ruby 2.7
+      #   return 1 # 1 is always ignored.
+      class TopLevelReturnWithArgument < Cop
+        MSG = 'Top level return with argument detected.'
+
+        def on_return(return_node)
+          add_offense(return_node) if ancestors_valid?(return_node) && return_node.arguments?
```

Comment:
I'd switch the conditions' order, as `arguments?` is very quick while `ancestors_valid?` might be slower

Reply:
Yes, this also makes a lot of sense given Ruby's short circuit evaluation.